src/avro/src/types.rs

```diff
@@ -8,18 +8,17 @@
 
 //! Logic handling the intermediate representation of Avro values.
 use std::collections::HashMap;
+use std::fmt;
 use std::hash::BuildHasher;
 use std::u8;
 
 use chrono::{NaiveDate, NaiveDateTime};
-use failure::Fail;
 use serde_json::Value as JsonValue;
 
 use crate::schema::{RecordField, SchemaNode, SchemaPiece, SchemaPieceOrNamed};
 
 /// Describes errors happened while performing schema resolution on Avro data.
-#[derive(Fail, Debug)]
-#[fail(display = "Schema resolution error: {}", _0)]
+#[derive(Debug)]
 pub struct SchemaResolutionError(pub String);
```

Comment:
I think that this is actually a pretty legitimate use for `thiserror`, but it's not necessary and probably doesn't gain anything until we start doing more structured things.

The thing that made me think of it is that deriving `thiserror::Error` allows specifying display with an attribute, but that _alone_ is obviously not worth the extra dep by itself, although I wouldn't be surprised if it's already in our tree for another reason. It's ability to add backtraces and causes is nice, though.

Reply:
Update: I checked and we are already compiling `thiserror` through transitive dependencies. I still have no opinion on if if it _should_ be used here, just updating.

Reply:
(Similar comment to below; I personally find `thiserror` to be way too magical, but we could take a poll, see what folks think, and enforce usage of the winning approach.)